khmer/_minhash.cc

```diff
@@ -0,0 +1,381 @@
+//
+// Python 2/3 compatibility: PyInt and PyLong
+//
+
+// Must be first.
+#include <Python.h>
+
+#if (PY_MAJOR_VERSION >= 3)
+#define PyInt_Check(arg) PyLong_Check(arg)
+#define PyInt_AsLong(arg) PyLong_AsLong(arg)
+#define PyInt_FromLong(arg) PyLong_FromLong(arg)
+#endif
+
+//
+// Python 2/3 compatibility: PyBytes and PyString
+// https://docs.python.org/2/howto/cporting.html#str-unicode-unification
+//
+
+#include "bytesobject.h"
+
+//
+// Python 2/3 compatibility: Module initialization
+// http://python3porting.com/cextensions.html#module-initialization
+//
+
+#if PY_MAJOR_VERSION >= 3
+#define MOD_ERROR_VAL NULL
+#define MOD_SUCCESS_VAL(val) val
+#define MOD_INIT(name) PyMODINIT_FUNC PyInit_##name(void)
+#define MOD_DEF(ob, name, doc, methods) \
+          static struct PyModuleDef moduledef = { \
+            PyModuleDef_HEAD_INIT, name, doc, -1, methods, }; \
+          ob = PyModule_Create(&moduledef);
+#else
+#define MOD_ERROR_VAL
+#define MOD_SUCCESS_VAL(val)
+#define MOD_INIT(name) void init##name(void)
+#define MOD_DEF(ob, name, doc, methods) \
+          ob = Py_InitModule3(name, methods, doc);
+#endif
+
+//
+// Function necessary for Python loading:
+//
+
+extern "C" {
+    MOD_INIT(_minhash);
+}
+
+
+#include <string>
+#include <map>
+#include <exception>
+#include <iostream>
+
+#include "kmer_min_hash.hh"
+
+////
+
+typedef struct {
+  PyObject_HEAD
+  KmerMinHash * mh;
+} MinHash_Object;
+
+static
+void
+MinHash_dealloc(MinHash_Object * obj)
+{
+  delete obj->mh;
+  obj->mh = NULL;
+  Py_TYPE(obj)->tp_free((PyObject*)obj);
+}
+
+static std::map<std::string, std::string> * _codon_table = NULL;
```

Comment:
Since we are using C++11 now, this can be simplified:

```c++
static std::map<std::string, std::string> _codon_table = {
       {"TTT", "F"}, {"TTC", "F"},
       {"TTA", "L"}, {"TTG", "L"},

       {"TCT", "S"}, {"TCC", "S"}, {"TCA", "S"}, {"TCG", "S"},

       {"TAT", "Y"}, {"TAC", "Y"},
       {"TAA", "*"}, {"TAG", "*"},

       {"TGT", "C"}, {"TGC", "C"},
       {"TGA", "*"},
       {"TGG", "W"},

       {"CTT", "L"}, {"CTC", "L"}, {"CTA", "L"}, {"CTG", "L"},

       {"CCT", "P"}, {"CCC", "P"}, {"CCA", "P"}, {"CCG", "P"},

       {"CAT", "H"}, {"CAC", "H"},
       {"CAA", "Q"}, {"CAG", "Q"},

       {"CGT", "R"}, {"CGC", "R"}, {"CGA", "R"}, {"CGG", "R"},

       {"ATT", "I"}, {"ATC", "I"}, {"ATA", "I"},
       {"ATG", "M"},

       {"ACT", "T"}, {"ACC", "T"}, {"ACA", "T"}, {"ACG", "T"},

       {"AAT", "N"}, {"AAC", "N"},
       {"AAA", "K"}, {"AAG", "K"},

       {"AGT", "S"}, {"AGC", "S"},
       {"AGA", "R"}, {"AGG", "R"},

       {"GTT", "V"}, {"GTC", "V"}, {"GTA", "V"}, {"GTG", "V"},

       {"GCT", "A"}, {"GCC", "A"}, {"GCA", "A"}, {"GCG", "A"},

       {"GAT", "D"}, {"GAC", "D"},
       {"GAA", "E"}, {"GAG", "E"},

       {"GGT", "G"}, {"GGC", "G"}, {"GGA", "G"}, {"GGG", "G"}
};
```

(See https://github.com/dib-lab/khmer/pull/1316/files#diff-709df73a4b93683a9c1dc0e8607f945aR68 for similar changes on HLL)